jumanji/environments/routing/multi_cvrp/types.py

```diff
@@ -0,0 +1,170 @@
+        start time it receives
+        pentality = (TimeWindow.start - Vehicle.local_times) * PenalityCoeff.early.
+    late: this array stores the late penalty coefficient values for
+        each customer. If a vehicle arrives after the customer's window
+        end time it receives
+        pentality = (Vehicle.local_times - TimeWindow.end) * PenalityCoeff.late.
+    """
+
+    early: chex.Array  # Shape: (num_customers,)
+    late: chex.Array  # Shape: (num_customers,)
+
+
+@dataclass
+class Vehicle:
+    """Vehicle tree structure.
+    local_times: This array stores the current local times of
+        each vehicle. This is necessary as each vehicle has traveled
+        a different total distance and therefore has different
+        local times for the same environment step.
+    capacities: This array stores the capacities of each vehicle. This
+        represents the total number of packages/volume that the vehicle can still
+        add before needing to return to the depot.
+    """
+
+    local_times: chex.Array  # Shape: (num_vehicles,)
+    capacities: chex.Array  # Shape: (num_vehicles,)
+
+
+@dataclass
+class ObsVehicle(Vehicle):
+    """Vehicle tree structure.
+    local_times: This array stores the current local times of
+        each vehicle. This is necessary as each vehicle has traveled
+        a different total distance and therefore has different
+        local times for the same environment step.
+    coordinates: This array stores the positions (locations) of each of
+        the vehicles.
+    capacities: This array stores the capacities of each vehicle. This
+        represents the total number of packages/volume that the vehicle can still
+        add before needing to return to the depot.
+    """
+
+    coordinates: chex.Array  # Shape: (num_vehicles, 2)
+
+
+@dataclass
+class StateVehicle(Vehicle):
+    """Vehicle tree structure.
+    local_times: This array stores the current local times of
+        each vehicle. This is necessary as each vehicle has traveled
+        a different total distance and therefore has different
+        local times for the same environment step.
+    positions: This array stores the positions (locations) of each of
+        the vehicles. Here 0 means that the vehicle is at the DEPOT and 1+ that
+        it is at a customer.
+    capacities: This array stores the capacities of each vehicle. This
+        represents the total number of packages/volume that the vehicle can still
+        add before needing to return to the depot.
+    distances: This array stores the total distances traveled by each vehicle
+        thus far.
+    time_penalties: This array stores the total time penalties that each vehicle
+        has received thus far.
+    """
+
+    positions: chex.Array  # Shape: (num_vehicles,)
+    distances: chex.Array = None  # Shape: (num_vehicles,)
+    time_penalties: chex.Array = None  # Shape: (num_vehicles,)
```

Comment:
Why having this default to None behaviour? It would mean that they are `Optional[chex.Array]`.
```suggestion
    distances: chex.Array  # Shape: (num_vehicles,)
    time_penalties: chex.Array  # Shape: (num_vehicles,)
```